src/json.cpp

```diff
@@ -1072,7 +1030,7 @@ static bool get_escaped_or_unicode( std::istream &stream, std::string &s, std::s
                         }
                     }
                     try {
-                        s += utf16_to_utf8( u );
+                        s += utf32_to_utf8( u );
```

Comment:
Works for dda, but breaks the formatter
```
g++-9 -ffast-math -Os -flto=jobserver -flto-odr-type-merging -Wodr -Werror -Wall -Wextra -Wformat-signedness -Wlogical-op -Wmissing-declarations -Wmissing-noreturn -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -Wpedantic -Wsuggest-override -Wunused-macros -Wzero-as-null-pointer-constant -Wno-unknown-warning-option -Wredundant-decls    -fsigned-char  -std=c++14 -MMD -MP -m64 -fuse-ld=gold -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -DCATA_IN_TOOL -Itools/format -Isrc \
  tools/format/format.cpp tools/format/format_main.cpp src/json.cpp -o tools/format/json_formatter.cgi
/tmp/json_formatter.cgi.OVk1Mk.ltrans0.ltrans.o:<artificial>:function JsonIn::get_string[abi:cxx11](): error: undefined reference to 'utf32_to_utf8[abi:cxx11](unsigned int)'
collect2: error: ld returned 1 exit status
make: *** [Makefile:1218: tools/format/json_formatter.cgi] Error 1
```

Reply:
OK, I've switched it to keep the version in `json.cpp` and remove the version in `catacharset.cpp` instead.